### PR TITLE
Progress hook

### PR DESCRIPTION
**Pull request by AMIRSOHAIL1998**

### CODE REVIEW

**Review by ShreyDhyani — CHANGES_REQUESTED** [review]
Pls make the requested changes.

File: toolkit/hooks/src/use-progress.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
+import React, { useState, useEffect, useRef } from "react";
+
+interface UseProgresInterface {
+    progressInterval?: number,
+    running?: true | false
```
[the remark this answers is not in the document]
Comment (AMIRSOHAIL1998) [reply toolkit/hooks/src/use-progress.ts]: done

File: toolkit/hooks/src/use-progress.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+      clearInterval(intervalRef.current as NodeJS.Timeout);
+    }
+  };
+
+  const stop = () => {
+    clearInterval(intervalRef.current as NodeJS.Timeout);
+  };
+
+  useEffect(() => {
+    if (running == true) {
```
[the remark this answers is not in the document]
Comment (AMIRSOHAIL1998) [reply toolkit/hooks/src/use-progress.ts]: used all the places

File: toolkit/hooks/src/use-progress.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
+import React, { useState, useEffect, useRef } from "react";
+
+interface UseProgresInterface {
```
[the remark this answers is not in the document]
Comment (AMIRSOHAIL1998) [reply toolkit/hooks/src/use-progress.ts:3]: added required type which is running only rest of the props are optional.

File: toolkit/hooks/src/useSlider.ts [hunk 4]
```diff
@@ -0,0 +1,119 @@
+import React, { useState, useEffect, useRef } from "react";
```
Comment (ShreyDhyani) [inline toolkit/hooks/src/useSlider.ts:1]: Please remove this file from Progress hook PR.

**Review by ShreyDhyani — CHANGES_REQUESTED** [review]

File: toolkit/hooks/src/useSlider.ts (hunk 4 above)
Comment (AMIRSOHAIL1998) [reply toolkit/hooks/src/useSlider.ts]: file s removed

File: toolkit/hooks/src/use-progress.ts [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+  useEffect(() => {
+    if (progress < 100 && totalFileSize !== 0) {
+      clearInterval(intervalRef.current as NodeJS.Timeout);
+      start();
+    }
+    const processedFileInPercent:number =
+      (processedFileSize / totalFileSize) * 100;
+    setCurrentFileProcessed(Number(processedFileInPercent.toFixed(0)));
+    console.log("processedFile", processedFileInPercent);
```
Comment (ShreyDhyani) [inline toolkit/hooks/src/use-progress.ts:87]: Remove this console.log

File: toolkit/hooks/src/use-progress.ts [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+    intervalRef.current = setInterval(() => {
+      updateProgress();
+    }, 10);
+  };
+
+  const pause = () => {
+    if (progress > 0 && progress < 100) {
+      const currentProgress:number = Number(currentProgressRef.current.toFixed(0));
+      setProgress(currentProgress);
+      clearInterval(intervalRef.current as NodeJS.Timeout);
```
Comment (ShreyDhyani) [inline toolkit/hooks/src/use-progress.ts:49]: NodeJS is used here but not imported anywhere in this file.

**Review by ShreyDhyani — CHANGES_REQUESTED** [review]

File: yarn.lock [hunk 7]
```diff
@@ -1265,116 +1265,6 @@
     human-id "^1.0.2"
```
[the remark this answers is not in the document]
Comment (AMIRSOHAIL1998) [reply yarn.lock]: taken latest pull and run yarn command is done

File: toolkit/hooks/src/use-progress.ts (hunk 5 above)
Comment (AMIRSOHAIL1998) [reply toolkit/hooks/src/use-progress.ts:87]: log is removed

File: toolkit/hooks/src/use-progress.ts (hunk 6 above)
Comment (AMIRSOHAIL1998) [reply toolkit/hooks/src/use-progress.ts:49]: this is added in tsconfig.json file that's why no need to import it ,automatically detects and not showing any error for that.

**Review by ShreyDhyani — APPROVED** [review]

**Pull request merged**